### ISSUE TITLE
MSW does not intercept axios request which has onUploadProgress configuration.

### ISSUE THREAD

**Issue by chetankondawle** [kettanaito/msw#1613]
### Discussed in https://github.com/mswjs/msw/discussions/1612

<div type='discussions-op-text'>
<br/>
<sup>Originally posted by **chetankondawle** May 10, 2023</sup>
<br/>
I have an file upload API, which has a config to check the progress of file being uplaoded using onUploadProgress. It takes a callback, which takes a progressEvent as ain argument, which gives me total file size and uploaded file size.
<br/>
<br/>
I tried mocking this particular upload api request with MSW. MSW does not intercept the request and directly the catch block is called in test case. When, I remvoe the onUploadProgress config, MSW intercepts the request and returns the resposne correctly.
<br/>
<br/>
How to mock the request which has onUploadProgress config using MSW?
<br/>
<br/>
Here is the code snippet for axios config.

```
axiosClient.put(
    `/uploadFile/files`,
    formData,
    {
        timeout: 35000,
        headers: {
            "Content-Type": "multipart/form-data",
        },
        onUploadProgress: (progressEvent) => {
            const progress: number = Math.round(
                (progressEvent.loaded * 100) / progressEvent.total
            );
            console.log("progress");
            setProgressMap((prev) => new Map(prev.set(fileName, progress)));
        },
    }).then((res) => {
        console.log("Here");
        return res.data;
    }).catch((error) => {
        console.log("error);
    });

```

Form data holds the file object .</div>

**Comment by RajB07** [comment]
Even I am facing the same issue. Any fixes for this @kettanaito ?

**Comment by kettanaito** [comment]
Hi. I've added Axios upload tests for both the browser and Node and they are passing without issues on MSW 2.0. Take a look at the PR to see the examples (#1855). I've also added a [File upload](https://mswjs.io/docs/recipes/file-uploads) recipe to the docs for reference.

**Comment by kettanaito** [comment]
## Released: v2.0.7 🎉

This has been released in v2.0.7!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.0.7)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.0.7)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
